src/compiler/program.ts

```diff
@@ -1832,7 +1833,7 @@ namespace ts {
         }
 
         function getBindAndCheckDiagnostics(sourceFile: SourceFile, cancellationToken?: CancellationToken): readonly Diagnostic[] {
-            return getBindAndCheckDiagnosticsForFile(sourceFile, cancellationToken);
+            return getBindAndCheckDiagnosticsForFile(sourceFile, cancellationToken, /*includeUncheckedJS*/ undefined);

```

Comment:
Having two different behavior between LS and command line could also mean that we cant use tsbuildinfo cached semantic diagnostics if and when we start using tsbuildinfo to report errors on closed files.

Reply:
I can think of a couple ways to fix this
- Always return all errors, then have `tsc` have a filter when displaying errors for JS files. This would add errors to all other API consumers, though.
- Give the language service a separate way to request unchecked JS errors.  The checker will have to store these errors separately and the language service will have to add them late.
- Possibly: Not report these errors in the "errors in closed files" list. But I don't know if that would be what users of that feature expect.

Opinions? Ideas?

I'm pretty sure that this feature is going to stick around and grow, so I'd like for it to be easier to use than to not use. Specifically, it's annoying if the future status quo for the language service is to request diagnostics twice, once for unchecked JS and again for everything else.

Reply:
> Give the language service a separate way to request unchecked JS errors. The checker will have to store these errors separately and the language service will have to add them late.

Isn't that what suggestion diagnostics are suppose to be and seems like a good place to add those errors ?

Reply:
They are quite similar, and that would work except that the display should be squiggly underlines, like an error, instead of three grey dots, like a suggestion.

Reply:
@mjbvz Would VS Code be OK to add a separate call to a new function `getUncheckedJSDiagnostics` with a matching protocol message `uncheckedJSDiagnosticsSync`? It would make it easier to create a setting to disable the errors and for Code to style them differently if desired.

Also @uniqueiniquity @jessetrinity you probably have opinions from the VS side.

Reply:
We certainly can add an additional call on the VS side.
I'm not sure I understand why it's inappropriate for `tsc` to report these, though?

Reply:
@uniqueiniquity is right, why should the behaviour be different between LS and tsc ?

Reply:
I do think treading lightly might be the way to go - we've never put errors in JS files unless they're specifically parsing errors.

Providing these as suggestion diagnostics keeps these as "quiet" by default. We can then potentially have nightly/insiders editors promote these to green or red squiggles (similar to how they specially-handle unused variable diagnostics) and see if we get any feedback on it. What do you all think? (CC @mjbvz @RyanCavanaugh)

Reply:
@uniqueiniquity @sheetalkamat 
> I'm not sure I understand why it's inappropriate for tsc to report these, though?

There are two scenarios here. The first is mixed TS/JS, where people have legacy JS as part of the compilation via allowJs. These people are running `tsc` regularly, and I don't want to block their build on a bogus error.

@DanielRosenwasser 
The other scenario is pure JS users who never run tsc. I believe it's these people that @DanielRosenwasser is thinking with
> I do think treading lightly might be the way to go

I agree. Shipping this as a suggestion at first is a good start. Based on the time I added all implicit any errors as suggestions, I think nobody on VS or Code will notice. Then styling them differently is a question of whether TS provides the errors in a separate function or whether the editor filters the suggestion list. I think either is fine but leaving it to the editors is certainly easier for me. =)